pints/_nested/__init__.py

```diff
@@ -780,13 +795,191 @@ def _update_logger(self):
                     self._next_message = self._message_interval * (
                         1 + self._i_message // self._message_interval)
+
+
+class Ellipsoid():
+    """
+    A class to represent N dimensional ellipsoids, which are needed by both
+    ellipsoidal nested sampling and MultiNest.
+
+    In "center form" the equation of an ellipsoid is given by:
+
+    ``(x-c).T * A * (x-c) = 1``
+
+    where ``A`` is a NxN dimensional positive definite symmetric matrix and
+    ``c`` is a N dimensional vector indicating the center of the ellipsoid.
+
+    Parameters
+    ----------
+    A : NxN dimensional positive definite symmetric matrix
+        represents the orientation and size of ellipsoid
+    c : N dimensional vector
+        center of ellipsoid
+    """
+    def __init__(self, A, c):
+
+        self._c = c
+        self._n_parameters = len(c)
+
+        if A.shape != (self._n_parameters, self._n_parameters):
+            raise ValueError(
+                'Sigma must have same dimension as mean, or be a square ' +
+                'matrix with the same dimension as the center.')
+        self._A = np.copy(A)
+
+        # calculate useful quantities
+        self._A_inv = np.linalg.inv(A)
+        # don't calculate volume unless needed
+        self._volume = None
+
+        # don't cache points unless constructed using minimum_volume_ellipsoid
+        self._points = None
+        self._n_points = 0
+
+    def __add__(self, other):
+        return [self, other]
+
+    def add_points(self, points):
+        """ Adds points contained within bounding ellispoid. """
+        self._points = points
+        self._n_points = len(points)
+
+    def centroid(self):
+        """ Returns centroid of ellispoid. """
+        return self._c
+
+    def enlarge(self, enlargement_factor):
+        """ Enlarges ellipsoid by a factor. """
+        self._A *= (1 / enlargement_factor)
+        self._A_inv *= enlargement_factor
+        self._volume = None
+
+    @staticmethod
+    def mahalanobis_distance(point, A, c):
+        """
+        Finds Mahalanobis distance between a point and the centroid of
+        of an ellipsoid.
+        """
+        return np.matmul(np.matmul(point - c, A), point - c)
+
+    @classmethod
+    def minimum_volume_ellipsoid(cls, points):
+        """
+        Creates an approximate minimum bounding ellipsoid in "center form":
+        ``(x-c).T * A * (x-c) = 1``.
+        """
+        cov = np.cov(np.transpose(points))
+        cov_inv = np.linalg.inv(cov)
+        c = np.mean(points, axis=0)
+        dist = np.zeros(len(points))
+        for i in range(len(points)):
+            dist[i] = Ellipsoid.mahalanobis_distance(points[i], cov_inv, c)
+        enlargement_factor = np.max(dist)
+        A = (1.0 / enlargement_factor) * cov_inv
+        obj = cls(A, c)
+        obj.add_points(points)
+        return obj
+
+    def n_points(self):
+        """ Returns number of points within bounding ellipsoid. """
```

Comment:
the

(few more of these)

Reply:
Not sure I get ya?

Reply:
the number of points within a(?) bounding ellipsoid

Reply:
I think this is ok, to be honest: it's more a personal style issue. (There are also examples like this abound throughout PINTS.)